### ISSUE TITLE
DG Correctness - What is considered repetitive? 

### ISSUE THREAD

**Issue by RunjiaChen** [nus-cs2103-AY2324S2/forum#745]
For my portion on the DG, I talked about my feature which prompts the user to enter the details of the contacts they want to add. 
I split the DG into stages of accepting different fields, as shown in the examples below 
<img width="187" alt="image" src="https://github.com/nus-cs2103-AY2324S2/forum/assets/142142406/653b4942-8a42-4313-b7ef-9892c1683ba6">

<img width="181" alt="image" src="https://github.com/nus-cs2103-AY2324S2/forum/assets/142142406/556d58c6-4241-49f0-9357-4f0b65bf1763">

The first diagram illustrates the flow of entering the email field, while the second diagram illustrates the flow of entering the name field. 

In the use case of the DG, I gave an example of the wrong input by the user, and and a correct input afterwards, then I used to diagram above to illustrate the sequence of the flow of the user actions

<img width="483" alt="image" src="https://github.com/nus-cs2103-AY2324S2/forum/assets/142142406/2297545a-ff10-4ecb-9b05-d24161ed3915">
<img width="483" alt="image" src="https://github.com/nus-cs2103-AY2324S2/forum/assets/142142406/08e9f1d6-c9d4-4209-abab-11a906bbce9d">

May I know if this may come off as too repetitive? The diagrams and the flow of actions are largely similar, but the type of 'wrong inputs' for name and email fields are fundamentally different.

**Comment by damithc** [comment]
What do others think?

**Comment by damithc** [comment]
@RunjiaChen To add, as this is a graded deliverable, I can't give direct comments, but it is fine to discuss among peers here.
I can add point to some relevant resources though:
* Tutorial 10 covered a discussion of possible DG bugs, which included an activity diagrams as well. The video is in https://nus-cs2103-ay2324s2.github.io/website/schedule/week10/tutorial.html 
Answer are in Canvas.
* The above page has a list of possible DG bugs and some DG tips as well 

A general comment: When a diagram is too big/complicated, we recommend breaking it down to smaller diagrams. But overdoing that will result in many small diagrams, which is not helpful either.

**Comment by baskargopinath** [comment]
maybe u can use a single diagram and change it to `User enters attribute` with a link so both name and email can point to the same diagram?

**Comment by PallonCX** [comment]
I agree with @bgopi23. To add on, I think it will be visually **repetitive** as a similar chunk of codes exists as multiple copies in the file (with minor changes), so it will be good if you can "abstract" their similarities out.
